modules/Sources/Dependencies/LogsHandler/LogsHandlerLive.swift

```diff
@@ -66,8 +73,16 @@ extension LogsHandlerClient: DependencyKey {
 }
 
 private extension LogsHandlerClient {
-    static func exportAndStoreLogsFor(key: String, atURL: URL) async throws -> (result: String, dir: URL) {
-        let logsStr = try await LogStore.exportCategory(key)
+    static func exportAndStoreLogsFor(
+        key: String,
+        atURL: URL,
+        level: OSLogEntryLog.Level = .debug
+    ) async throws -> (result: String, dir: URL) {
+        let logsStr = try await LogStore.exportCategory(
+            key,
+            level: level,
+            fileSize: 2_000_000 // ~ 2MB of data
```

Comment:
Does this limit the result log size? Or how can we ensure that the logs fit into the file size?

Reply:
The function `exportCategory` takes all available logs that match the criteria (category, level, bundle, ...) and cycle through it. Counts the size of every log in bytes and append only up to the given fileSize.